{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "de0d7600",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "A module that was compiled using NumPy 1.x cannot be run in\n",
      "NumPy 2.0.2 as it may crash. To support both 1.x and 2.x\n",
      "versions of NumPy, modules must be compiled with NumPy 2.0.\n",
      "Some module may need to rebuild instead e.g. with 'pybind11>=2.12'.\n",
      "\n",
      "If you are a user of the module, the easiest solution will be to\n",
      "downgrade to 'numpy<2' or try to upgrade the affected module.\n",
      "We expect that some modules will need time to support NumPy 2.\n",
      "\n",
      "Traceback (most recent call last):  File \"/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/runpy.py\", line 196, in _run_module_as_main\n",
      "    return _run_code(code, main_globals, None,\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/runpy.py\", line 86, in _run_code\n",
      "    exec(code, run_globals)\n",
      "  File \"/Users/atiehameri/Documents/Master/02_CIS_602/01_Skin Cancer Classification/Skin_Scan_App_Django/.venv/lib/python3.10/site-packages/ipykernel_launcher.py\", line 18, in <module>\n",
      "    app.launch_new_instance()\n",
      "  File \"/Users/atiehameri/Documents/Master/02_CIS_602/01_Skin Cancer Classification/Skin_Scan_App_Django/.venv/lib/python3.10/site-packages/traitlets/config/application.py\", line 1075, in launch_instance\n",
      "    app.start()\n",
      "  File \"/Users/atiehameri/Documents/Master/02_CIS_602/01_Skin Cancer Classification/Skin_Scan_App_Django/.venv/lib/python3.10/site-packages/ipykernel/kernelapp.py\", line 739, in start\n",
      "    self.io_loop.start()\n",
      "  File \"/Users/atiehameri/Documents/Master/02_CIS_602/01_Skin Cancer Classification/Skin_Scan_App_Django/.venv/lib/python3.10/site-packages/tornado/platform/asyncio.py\", line 205, in start\n",
      "    self.asyncio_loop.run_forever()\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/asyncio/base_events.py\", line 603, in run_forever\n",
      "    self._run_once()\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/asyncio/base_events.py\", line 1909, in _run_once\n",
      "    handle._run()\n",
      "  File \"/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/asyncio/events.py\", line 80, in _run\n",
      "    self._context.run(self._callback, *self._args)\n",
      "  File \"/Users/atiehameri/Documents/Master/02_CIS_602/01_Skin Cancer Classification/Skin_Scan_App_Django/.venv/lib/python3.10/site-packages/ipykernel/kernelbase.py\", line 545, in dispatch_queue\n",
      "    await self.process_one()\n",
      "  File \"/Users/atiehameri/Documents/Master/02_CIS_602/01_Skin Cancer Classification/Skin_Scan_App_Django/.venv/lib/python3.10/site-packages/ipykernel/kernelbase.py\", line 534, in process_one\n",
      "    await dispatch(*args)\n",
      "  File \"/Users/atiehameri/Documents/Master/02_CIS_602/01_Skin Cancer Classification/Skin_Scan_App_Django/.venv/lib/python3.10/site-packages/ipykernel/kernelbase.py\", line 437, in dispatch_shell\n",
      "    await result\n",
      "  File \"/Users/atiehameri/Documents/Master/02_CIS_602/01_Skin Cancer Classification/Skin_Scan_App_Django/.venv/lib/python3.10/site-packages/ipykernel/ipkernel.py\", line 362, in execute_request\n",
      "    await super().execute_request(stream, ident, parent)\n",
      "  File \"/Users/atiehameri/Documents/Master/02_CIS_602/01_Skin Cancer Classification/Skin_Scan_App_Django/.venv/lib/python3.10/site-packages/ipykernel/kernelbase.py\", line 778, in execute_request\n",
      "    reply_content = await reply_content\n",
      "  File \"/Users/atiehameri/Documents/Master/02_CIS_602/01_Skin Cancer Classification/Skin_Scan_App_Django/.venv/lib/python3.10/site-packages/ipykernel/ipkernel.py\", line 449, in do_execute\n",
      "    res = shell.run_cell(\n",
      "  File \"/Users/atiehameri/Documents/Master/02_CIS_602/01_Skin Cancer Classification/Skin_Scan_App_Django/.venv/lib/python3.10/site-packages/ipykernel/zmqshell.py\", line 549, in run_cell\n",
      "    return super().run_cell(*args, **kwargs)\n",
      "  File \"/Users/atiehameri/Documents/Master/02_CIS_602/01_Skin Cancer Classification/Skin_Scan_App_Django/.venv/lib/python3.10/site-packages/IPython/core/interactiveshell.py\", line 3077, in run_cell\n",
      "    result = self._run_cell(\n",
      "  File \"/Users/atiehameri/Documents/Master/02_CIS_602/01_Skin Cancer Classification/Skin_Scan_App_Django/.venv/lib/python3.10/site-packages/IPython/core/interactiveshell.py\", line 3132, in _run_cell\n",
      "    result = runner(coro)\n",
      "  File \"/Users/atiehameri/Documents/Master/02_CIS_602/01_Skin Cancer Classification/Skin_Scan_App_Django/.venv/lib/python3.10/site-packages/IPython/core/async_helpers.py\", line 128, in _pseudo_sync_runner\n",
      "    coro.send(None)\n",
      "  File \"/Users/atiehameri/Documents/Master/02_CIS_602/01_Skin Cancer Classification/Skin_Scan_App_Django/.venv/lib/python3.10/site-packages/IPython/core/interactiveshell.py\", line 3336, in run_cell_async\n",
      "    has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n",
      "  File \"/Users/atiehameri/Documents/Master/02_CIS_602/01_Skin Cancer Classification/Skin_Scan_App_Django/.venv/lib/python3.10/site-packages/IPython/core/interactiveshell.py\", line 3519, in run_ast_nodes\n",
      "    if await self.run_code(code, result, async_=asy):\n",
      "  File \"/Users/atiehameri/Documents/Master/02_CIS_602/01_Skin Cancer Classification/Skin_Scan_App_Django/.venv/lib/python3.10/site-packages/IPython/core/interactiveshell.py\", line 3579, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"/var/folders/jx/9sgnmn655494mw2_gc1yt6hm0000gn/T/ipykernel_47084/2195286300.py\", line 54, in <module>\n",
      "    results = load_and_predict(\"model.tflite\", \"aug_ISIC_0082829_0.jpg\")\n",
      "  File \"/var/folders/jx/9sgnmn655494mw2_gc1yt6hm0000gn/T/ipykernel_47084/2195286300.py\", line 12, in load_and_predict\n",
      "    interpreter = Interpreter(\n",
      "  File \"/Users/atiehameri/Documents/Master/02_CIS_602/01_Skin Cancer Classification/Skin_Scan_App_Django/.venv/lib/python3.10/site-packages/tflite_runtime/interpreter.py\", line 348, in __init__\n"
     ]
    },
    {
     "ename": "AttributeError",
     "evalue": "_ARRAY_API not found",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
      "\u001b[0;31mAttributeError\u001b[0m: _ARRAY_API not found"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Error: Didn't find op for builtin opcode 'FULLY_CONNECTED' version '12'. An older version of this builtin might be supported. Are you using an old TFLite binary with a newer model?\n",
      "Registration failed.\n",
      "\n"
     ]
    }
   ],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": ".venv",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
